### PR TITLE
testnet_v1: bump to Godwoken v1.12.0-rc2-smt-trie and Web3/Indexer 1.12-rc

### PR DESCRIPTION
**Pull request by Simon-Tl**
## Notable Changes
- Highlights
   - Builtin mainnet and testnet consensus config https://github.com/godwokenrises/godwoken/pull/946
      The consensus-related options were moved into consensus config, and the builtin testnet and mainnet consensus were put into the godwoken binary. This change reduces the operation works of node maintainers

      **~200 lines of Godwoken node [config](https://github.com/godwokenrises/godwoken-info/blob/8eea5d5925d03b0c64eadf23cca40b267f085ef4/testnet_v1_1/gw-testnet_v1-config-readonly.toml) was reduced,** from 284 lines to 84 lines.

- Enhancement:
  - chore: use TransactionDB and refactor store https://github.com/godwokenrises/godwoken/pull/903
  - chore: refactor RPC server https://github.com/godwokenrises/godwoken/pull/927
  - Improve the error code of API when executing transactions https://github.com/godwokenrises/godwoken/pull/930

- Note:
   - To enable SMT-trie feature, data migration is required for this upgrade.  It takes about 75 minutes
      ```bash
        # migrate command for SMT trie feature, only needs to be run on first activation
        godwoken migrate --config /deploy/config.toml | tee -a /mnt/smt-migrate.log
      ``` 
  
## Detailed Release notes
   - Godwoken
      https://github.com/godwokenrises/godwoken/releases

  - Godwoken-web3
     https://github.com/godwokenrises/godwoken-web3/releases
  

## Changed Configs
1.  introduced testnet builtin consensus config
   ```TOML
    # builtin consensus config
    # see https://github.com/godwokenrises/godwoken/pull/946
    [consensus]
    # https://github.com/godwokenrises/godwoken/blob/develop/crates/config/src/consensus/builtins/testnet.toml
    builtin = 'Testnet'
   ```

2. [[dynamic_config.fee_config] -> [mem_pool.fee]](https://github.com/godwokenrises/godwoken-info/pull/88/files#diff-7e686c21daa843938ca4c0df5db2fe3b79bdb6e1845d19659f941f0a811e0976R76)

3. [[dynamic_config.rpc_config] -> [mem_pool.extra]](https://github.com/godwokenrises/godwoken-info/pull/88/files#diff-7e686c21daa843938ca4c0df5db2fe3b79bdb6e1845d19659f941f0a811e0976R82)

### CODE REVIEW

File: testnet_v1_1/gw-testnet_v1-config-readonly.toml [hunk 1]
```diff
@@ -206,79 +68,17 @@ max_cycles_limit = '1950000000'
 # https://github.com/godwokenrises/godwoken/blob/v1.7.0/crates/config/src/config.rs#L579-L599
 
 [store]
-path = '/mnt/store-20221008.db'
+# The data directory of the existing node is replaced
+path = '/mnt/testnet-20230215.rocksdb'
```
Comment (Flouse) [inline testnet_v1_1/gw-testnet_v1-config-readonly.toml]:
```suggestion
path = '/mnt/store-20221008.db'
```

File: testnet_v1_1/docker-compose.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ services:
   # see: https://docs.nervos.org/docs/basics/guides/run-ckb-with-docker#run-a-ckb-testnet-node
   gw-readonly:
     container_name: gw-testnet_v1-readonly
-    image: ghcr.io/godwokenrises/godwoken:v1.8.0-rc3
+    image: ghcr.io/godwokenrises/godwoken:1.12-rc-smt-trie
```
Comment (Flouse) [inline testnet_v1_1/docker-compose.yml]:
```suggestion
    image: ghcr.io/godwokenrises/godwoken:v1.12.0-rc2-smt-trie
```

File: testnet_v1_1/docker-compose.yml [hunk 3]
```diff
@@ -29,8 +29,14 @@ services:
...
-        godwoken rewind-to-last-valid-block --config-path /deploy/config.toml
-        godwoken run -c /deploy/config.toml
+        # migrate command for SMT trie feature, only needs to be run on first activation
+        godwoken migrate --config /deploy/config.toml | tee -a /mnt/smt-migrate.log
+        # use this command to rewind the chain to the last valid block, if the node has encountered
+        # some bad blocks accidentally due to e.g. version/configuration differences.
+        # see also: https://github.com/godwokenrises/godwoken/pull/832
+        # godwoken rewind-to-last-valid-block --config-path /deploy/config.toml
+        #migrated execute
+        #godwoken run -c /deploy/config.toml
```
Comment (Flouse) [inline testnet_v1_1/docker-compose.yml]:
```suggestion

        # Run Godwoken
        godwoken run -c /deploy/config.toml
```

**Review by Flouse — CHANGES_REQUESTED** [review]

**Review by jjyr — APPROVED** [review]

**Review by Flouse — APPROVED** [review]

**Comment by Flouse** [comment]
The gw-readonly-node of GwScan has been upgraded to v1.12.0-rc2-smt-trie

**Pull request merged**